### ISSUE TITLE
SSH exploiter hangs

### ISSUE THREAD

**Issue by shreyamalviya** [guardicore/monkey#2589]
## Describe the bug
The Agent hangs after it tries to disconnect (maybe in other cases as well, haven't investigated) a user from a relay. Happens both, from binary and from source.
See logs for more. Scroll down to the end and notice the timestamps. 

## To Reproduce
1. Configure the Monkey to run tunneling tests.
2. Wait and see logs to see that the Agent hangs.

## Expected behavior
The Agent shouldn't hang.

## Logs
[2022-11-16T11.47.19.810Z-island-linux-250__tunneling_hangs_binary.log](https://github.com/guardicore/monkey/files/10021417/2022-11-16T11.47.19.810Z-island-linux-250__tunneling_hangs_binary.log)

**Edit:** Not sure what's up. Here's another log file that doesn't make it seem like it's the relay logic's fault.
[2022-11-17T10.30.29.590Z-island-linux-250.log](https://github.com/guardicore/monkey/files/10030681/2022-11-17T10.30.29.590Z-island-linux-250.log)

## Tasks

- [ ] Fix it (0.75d)

**Comment by mssalvatore** [comment]
@shreyamalviya I ran this 6 or 7 times and was unable to reproduce it. Are there any other details you can share that will allow us to reproduce it more reliably? Which environment did you use?

**Comment by shreyamalviya** [comment]
I ran it again on the test-1 project's zoo today with the configuration of the depth_4_a test and encountered it. I was running it with depth_3_a yesterday. Ran the agent from source on the island-linux-250 machine.

**Comment by shreyamalviya** [comment]
The issue was with my account on our GCP projects. With the help of https://github.com/paramiko/paramiko/issues/2147, I found that running [`ssh-agent`](https://rabexc.org/posts/using-ssh-agent) and then running the SSH exploiter resulted in it working. Turns out the issue was with `ssh-add -l` which is used to list out all identities that `ssh-agent` has access to. With `ssh-agent` not running, it was giving me an invalid format error instead of saying that it couldn't connect to the agent. If I ran `ssh-agent` and then killed it, it worked as expected, until the next reboot. Tested this with a paramiko script too and it works.

Using the changes in #2612, the exploiter passes. Since our SSH exploiter uses credentials that it's given already, `ssh-agent` adds no value, so these changes won't affect it, but it'll make sure that the exploiter doesn't hang for a user because of issues with `ssh-agent`.

There could be a possible use case for the `ssh-agent` in exploitation. [Here's an issue](https://github.com/guardicore/monkey/issues/2614) to track that.
